### ISSUE TITLE
Compiling AesGcmJce for Java requires Android SDK

### ISSUE THREAD

**Issue by ipetr0v** [google/tink#507]
**Help us help you**

Please tell us more about [your Tink deployment](https://docs.google.com/forms/d/1mhHvyNJQgTXFDnqOermB7-BD8GQSyvtFPUAhILRbYcA/edit).

**Describe the bug**

I'm trying to use AES-GCM Tink implementation in Java (**not** Android). And compile it with Bazel using the following dependency `"@tink_java//src/main/java/com/google/crypto/tink/subtle:aes_gcm_jce"`.

But Bazel wants me to use Android SDK (though I don't try to compile it for Android).
```
ERROR: While resolving toolchains for target @tink_java//src/main/java/com/google/crypto/tink/config:tink_fips-android: no matching toolchains found for types @bazel_tools//tools/android:sdk_toolchain_type
```
Same thing is happening for `AesGcmHkdf`

**To Reproduce**

`BUILD` file
```Skylark
load("@rules_java//java:defs.bzl", "java_binary")

java_binary(
    name = "bugreport",
    srcs = [
        "com/bugreport/Main.java",
    ],
    main_class = "com.bugreport.Main",
    deps = [
        "@tink_java//src/main/java/com/google/crypto/tink/subtle:aes_gcm_jce",
    ],
)
```
Main Java file:
```java
package com.bugreport;

import com.google.crypto.tink.subtle.AesGcmJce;

public class Main {
    private AesGcmJce encryptor;
    public static void main(String[] args) {}
}
```

**Expected behavior**

It should build Tink for Java and not use any `android_library` rules.

**Error messages, stack traces, etc.**

```
ERROR: While resolving toolchains for target @tink_java//src/main/java/com/google/crypto/tink/config:tink_fips-android: no matching toolchains found for types @bazel_tools//tools/android:sdk_toolchain_type
```

**Version information**

 - Language: Java
 - Version: 1.6.0
 - Environment: Bazel, JDK 8

**Additional context**

It looks like some Java libraries require `tink_fips-android` instead of just `tink_fips`:
https://github.com/google/tink/blob/f874c4eb7b8610860fb63d4d53be0f9f524c1161/java_src/src/main/java/com/google/crypto/tink/subtle/BUILD.bazel#L191-L198

**Comment by ipetr0v** [comment]
This patch solves the issue locally:
```
diff --git src/main/java/com/google/crypto/tink/subtle/BUILD.bazel src/main/java/com/google/crypto/tink/subtle/BUILD.bazel
index a9fde1379..59d97ef12 100644
--- src/main/java/com/google/crypto/tink/subtle/BUILD.bazel
+++ src/main/java/com/google/crypto/tink/subtle/BUILD.bazel
@@ -193,7 +193,7 @@ java_library(
     srcs = ["Validators.java"],
     deps = [
         ":enums",
-        "//src/main/java/com/google/crypto/tink/config:tink_fips-android",
+        "//src/main/java/com/google/crypto/tink/config:tink_fips",
     ],
 )
```
Note: paths have `java_src` removed, because Bazel Workspace looks like this:
```
# Tink crypto library for Java.
http_archive(
    name = "tink_java",
    sha256 = "5856b0207ffb2cf28dd5c421789ffca3cfeea0680055f455e14bec2f335b1765",
    strip_prefix = "tink-58be99b3c4d09154d12643327f293cc45b2a6a7b/java_src",
    # Commit from 2021-05-19
    urls = [
        "https://github.com/google/tink/archive/58be99b3c4d09154d12643327f293cc45b2a6a7b.tar.gz",
    ],
    patches = [
        # This patch removes Android dependencies from Tink Java libraries.
        # https://github.com/google/tink/issues/507
        "//third_party/google/tink:Remove-android-from-java.patch",
    ],
)
```

**Comment by kste** [comment]
Thank you for reporting this. I already patched this as you described and should be up on Github pretty soon :-).

**Comment by chuckx** [comment]
Fixed in 6ed3bf03d1cdc19329c1908ee352137ba8c5427d.
